### ISSUE TITLE
list,tuple type assign to protocol type with warning

### ISSUE THREAD

**Issue by cole-dda** [microsoft/pyright#7925]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**
list,tuple type assign to protocol type with warning

**Code or Screenshots**
<img width="511" alt="screenshot_6399" src="https://github.com/microsoft/pyright/assets/1077830/f040fee3-7b5b-4d7d-8bdf-e1549e763ea7">

```python
@runtime_checkable
class A[T:int|float](Protocol):

    @overload
    def __getitem__(self,index:int)->T:
        ...
    
    @overload
    def __getitem__(self,index:slice)->Sequence[T]:
        ...

    def __getitem__(self,index:int|slice)->T|Sequence[T]:
        ...
    
def test(a:A[int]):
    pass

def usage():
    #pyright show waring
    test([1,2,3])
    
    #pyright ok
    test(cast(Sequence[int],[1,2,3]))
```

**VS Code extension or command-line**
vscode 1.89.1

**Comment by LeeeeT** [comment]
In typeshed the first argument to `list.__getitem__` is [defined](https://github.com/python/typeshed/blob/main/stdlib/builtins.pyi#L1001-L1004) as positional-only. That's why `list` isn't compatible with your protocol. To make it compatible consider making `index` positional-only like this:

```python
@runtime_checkable
class A[T: int | float](Protocol):
    @overload
    def __getitem__(self, index: int, /) -> T: ...

    @overload
    def __getitem__(self, index: slice, /) -> Sequence[T]: ...

    def __getitem__(self, index: int | slice, /) -> T | Sequence[T]: ...
```
